File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    
    var alphabet = [];
    
    for(let i=97; i<=122; i++){
        alphabet.push(String.fromCharCode(i));
    }
    
    alphabet = alphabet.filter((char) => !skip.includes(char));
    
    var answer = '';
    
    for(let i=0; i<s.length; i++){
        var idx = alphabet.indexOf(s.charAt(i));
        idx = (idx + index)%alphabet.length;
        answer += alphabet[idx];
    }
    
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        
        ArrayList<Character> alphabet = new ArrayList<>();
        
        for(int i=97; i<=122; i++){
            alphabet.add((char)i);
        }
        
        for(int i=0; i<skip.length(); i++){
            int idx = alphabet.indexOf(skip.charAt(i));
            alphabet.remove(idx);
        }
        
        String answer = "";
        
        int size = alphabet.size();
        
        for(int i=0; i<s.length(); i++){
            int idx = alphabet.indexOf(s.charAt(i));
            idx = (idx+index)%size;
            answer += alphabet.get(idx);
        }
        
        return answer;
    }
}